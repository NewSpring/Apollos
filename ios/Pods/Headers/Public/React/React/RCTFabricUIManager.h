../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h