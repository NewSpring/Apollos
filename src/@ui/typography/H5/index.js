import { Platform, Text } from 'react-native';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import styled from '@ui/styled';
import { withPlaceholder, Typography } from '@ui/Placeholder';

const styles = styled(
  ({ theme }) => ({
    fontSize: theme.helpers.rem(1.112),
    fontFamily: theme.typography.fontFamilySans.bold.default,
    color: theme.colors.text.primary,
    ...Platform.select({
      ios: {
        paddingTop: theme.helpers.rem(0.25),
        paddingBottom: theme.helpers.rem(0.175),
        lineHeight: theme.helpers.verticalRhythm(1.112, 1.13889),
      },
      android: {
        paddingTop: theme.helpers.rem(0.1),
        lineHeight: theme.helpers.verticalRhythm(1.112, 1.34),
      },
      web: {
        paddingTop: theme.helpers.rem(0.325),
        paddingBottom: theme.helpers.rem(0.1),
        lineHeight: theme.helpers.verticalRhythm(1.112, 1.13889),
        WebkitFontSmoothing: 'antialiased',
        MozOsxFontSmoothing: 'grayscale',
      },
    }),
  }),
  'H5',
);

const H5 = compose(
  setPropTypes({
    isLoading: PropTypes.bool, // display loading placeholder
    ...Text.propTypes,
  }),
  styles,
  withPlaceholder(Typography, { width: '60%' }),
  pure,
)(Text);

export default H5;
