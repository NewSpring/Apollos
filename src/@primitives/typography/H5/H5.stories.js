import React from 'react';

import { storiesOf } from '@storybook/react-native';

import ThemeProvider from '@primitives/ThemeProvider';
import H5 from './';

storiesOf('@primitives/typography', module)
  .add('H5', () => (
    <ThemeProvider>
      <H5>{'"Faith does not eliminate questions. But faith knows where to take them." â€• Elisabeth Elliot'}</H5>
    </ThemeProvider>
  ));
