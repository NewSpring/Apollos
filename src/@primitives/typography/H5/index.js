import React from 'react';
import { Text } from 'react-native';
import { compose, pure, setPropTypes } from 'recompose';
import PropTypes from 'prop-types';
import styled from '@primitives/styled';
import rem from '@utils/remUnit';
import verticalRhythm from '@utils/verticalRhythm';

const enhance = compose(
  pure,
  setPropTypes({
    children: PropTypes.node,
    style: Text.propTypes.style,
  }),
);

const StyledH5 = styled(({ theme }) => ({
  fontSize: rem(1.112, theme),
  lineHeight: verticalRhythm(1.112, 1.145, theme),
  fontWeight: '700',
  color: theme.baseFontColor,
}))(Text);

const H5 = enhance(({
  children,
  style: styleProp = {},
  ...otherProps
}) => (
  <StyledH5
    style={styleProp}
    {...otherProps}
  >
    {children}
  </StyledH5>
));

export default H5;
