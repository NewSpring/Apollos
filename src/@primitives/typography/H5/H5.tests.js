import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from '@primitives/ThemeProvider';
import H5 from './';

describe('the H5 component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <H5>Default H5 text</H5>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmon = { color: 'salmon' };
    const tree = renderer.create(
      <ThemeProvider>
        <H5 style={salmon}>Salmon text</H5>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept additional props', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <H5 accessible={false}>{'"Faith does not eliminate questions. But faith knows where to take them." â€• Elisabeth Elliot'}</H5>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
