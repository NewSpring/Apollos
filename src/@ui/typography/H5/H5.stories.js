import React from 'react';

import { storiesOf } from '@storybook/react-native';

import ThemeProvider from '@ui/ThemeProvider';
import H5 from './';

storiesOf('@ui/typography', module)
  .add('H5', () => (
    <ThemeProvider>
      <H5>{'"Faith does not eliminate questions. But faith knows where to take them." â€• Elisabeth Elliot'}</H5>
    </ThemeProvider>
  ));
