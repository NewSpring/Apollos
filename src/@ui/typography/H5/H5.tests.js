import React from 'react';
import renderer from 'react-test-renderer';
import Providers from '@ui/TestProviders';
import H5 from './';

describe('the H5 component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Providers>
        <H5>Default H5 text</H5>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmon = { color: 'salmon' };
    const tree = renderer.create(
      <Providers>
        <H5 style={salmon}>Salmon text</H5>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept additional props', () => {
    const tree = renderer.create(
      <Providers>
        <H5 accessible={false}>{'"Faith does not eliminate questions. But faith knows where to take them." â€• Elisabeth Elliot'}</H5>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
