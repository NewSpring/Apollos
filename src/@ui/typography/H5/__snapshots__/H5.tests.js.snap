// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the H5 component should accept additional props 1`] = `
<Text
  accessible={false}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#303030",
      "fontFamily": "Colfax-Bold",
      "fontSize": 20.02,
      "lineHeight": 25.33,
      "paddingBottom": 3.15,
      "paddingTop": 4.5,
    }
  }
>
  "Faith does not eliminate questions. But faith knows where to take them." â€• Elisabeth Elliot
</Text>
`;

exports[`the H5 component should accept and render passed in styles 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "salmon",
      "fontFamily": "Colfax-Bold",
      "fontSize": 20.02,
      "lineHeight": 25.33,
      "paddingBottom": 3.15,
      "paddingTop": 4.5,
    }
  }
>
  Salmon text
</Text>
`;

exports[`the H5 component should render correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#303030",
      "fontFamily": "Colfax-Bold",
      "fontSize": 20.02,
      "lineHeight": 25.33,
      "paddingBottom": 3.15,
      "paddingTop": 4.5,
    }
  }
>
  Default H5 text
</Text>
`;
